osidb/api_views.py

```diff
@@ -1717,3 +1721,445 @@ def post(self, request, *args, **kwargs):
+        """List all sync managers with optional filtering"""
+
+        # Get all SyncManager subclasses
+        manager_classes = SyncManager.__subclasses__()
+
+        # Collect filtered querysets from all subclasses
+        querysets = []
+        for manager_class in manager_classes:
+            queryset = manager_class.objects.all()
+
+            # Apply filters
+            manager_type_filter = request.query_params.get("manager_type")
+            if manager_type_filter:
+                manager_types = [t.strip() for t in manager_type_filter.split(",")]
+                if manager_class.__name__ not in manager_types:
+                    continue
+
+            sync_id_filter = request.query_params.get("sync_id")
+            if sync_id_filter:
+                sync_ids = [s.strip() for s in sync_id_filter.split(",")]
+                queryset = queryset.filter(sync_id__in=sync_ids)
+
+            permanently_failed_filter = request.query_params.get("permanently_failed")
+            if (
+                permanently_failed_filter is not None
+                and permanently_failed_filter != ""
+            ):
+                permanently_failed_value = strtobool(permanently_failed_filter)
+                queryset = queryset.filter(permanently_failed=permanently_failed_value)
+
+            # Apply DateTime filters for all DateTime fields
+            datetime_fields = [
+                "last_scheduled_dt",
+                "last_started_dt",
+                "last_finished_dt",
+                "last_failed_dt",
+                "last_rescheduled_dt",
+            ]
+            for field_name in datetime_fields:
+                queryset = self._apply_datetime_filters(queryset, request, field_name)
+
+            # Annotate queryset with manager_type to identify the subclass
+            queryset = queryset.annotate(manager_type=Value(manager_class.__name__))
+
+            querysets.append(queryset)
+
+        # Combine all querysets using union
+        if not querysets:
+            # No manager classes found, return empty paginated response
+            paginator = self.pagination_class()
+            return paginator.get_paginated_response([])
+
+        # Union all querysets SyncManager subclasses querysets
+        # all=True to allow duplicates.
+        combined_queryset = querysets[0].union(*querysets[1:], all=True)
+
+        # Apply pagination
+        paginator = self.pagination_class()
+        page = paginator.paginate_queryset(combined_queryset, request)
+
+        if page is not None:
+            # Serialize paginated results
+            serializer = SyncManagerSerializer(page, many=True)
+            return paginator.get_paginated_response(serializer.data)
+
+        # Fallback if pagination is not applied (shouldn't happen with pagination_class set)
+        serializer = SyncManagerSerializer(combined_queryset, many=True)
+        return Response(serializer.data)
```

Comment:
All the filtering should be done in the [`filters.py` ](https://github.com/RedHatProductSecurity/osidb/blob/master/osidb/filters.py) module and leverage `django-filter` library.